mesh/mesh.hpp

```diff
@@ -57,11 +58,16 @@ class Mesh
 #endif
    friend class NCMesh;
    friend class NURBSExtension;
+   friend class MeshConnections;
 
 #ifdef MFEM_USE_ADIOS2
    friend class adios2stream;
 #endif
 
+public:
+
+   MeshConnections connect;
```

Comment:
I was thinking that this object would be created by the users when needed:
```c++
const Mesh mesh(...);
const MeshConnections mesh_connections(mesh);
```
But composition can work too...

Reply:
Yeah, I want to put it here and deprecate the existing interfaces to the tables in Mesh.